File: 04_09/I.py

if __name__ == '__main__':
    n = int(input())
    count = 0
    for a in range(1, n):
        for b in range(a, n):
            c = __import__('math').sqrt((a * a) + (b * b))
            if(int(c) == c and c <= n):
                count += 1
            elif(c > n):
                break
    print(count)

File: 04_09/I.cpp
#include <iostream>
#include <math.h>
using namespace std;

int n, count; 
int main (){
	cin >> n;
    for (int a = 1; a < n; a++){
        for (int b = a; b < n; b++){
            int aux  = ((a * a) + (b * b));
            int c = sqrt(aux);
            if((c*c == (aux)) && (c <= n)){
                count += 1;
            }else if(c > n)
                break;
        }
    }
    cout << count << endl;
}
